org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionHandlerTest.java

```diff
@@ -237,7 +237,8 @@ public void testCompletion_object() throws Exception{
 			assertTrue(isNotBlank(item.getInsertText()));
 			assertNotNull(item.getFilterText());
 			assertFalse(item.getFilterText().contains(" "));
-			assertTrue(item.getLabel().startsWith(item.getFilterText()));
+			assertTrue(item.getLabel().startsWith(item.getInsertText()));
```

Comment:
This case was to ensure that `label` is of the format "SimpleName - PackageName", nothing to do with `filterText`. Now since `InsertText` is set to the SimpleName as a fallback, I'm simply using it instead. 

In the below line, I add a case to ensure `filterText` starts with the **prefix**, which is calculated from `textEdit.range`